### ISSUE TITLE
PIE794 false positive when mutating class variable. Also leads to a bad fix.

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#8497]
```python
class Person:
    name = "Foo"
    name = name + " Bar"
```
with unsafe fixes applied becomes:
```python
class Person:
    name = "Foo"
```
It may not be the cleanest way of writing that, but it probably shouldn't trigger.

```python
class Person:
    names = ["Foo"]
    names = names + ["Bar"]
```
is a use case I've actually seen in real code that probably should not be flagged.

ruff version is 0.1.4.

**Comment by Skylion007** [comment]
making it do `+=` seems to be a decent mitigation for the objects which define it, but that may not always be an option depending on the Python object.
